odml/doc.py

```diff
@@ -32,7 +32,7 @@ def __init__(self, author=None, date=None, version=None, repository=None, id=Non
                 self._id = str(uuid.uuid4())
         except ValueError as e:
             print(e)
-            self._id = str(uuid.uuid4())
+        self._id = str(uuid.uuid4())
```

Comment:
This currently always overwrites any previously set id with a random one rendering the `try ... catch` ineffectual.